controllers/storageoscluster/node-manager_operand.go

```diff
@@ -227,12 +222,22 @@ func (c *NodeManagerOperand) Ensure(ctx context.Context, obj client.Object, owne
 	}
 
 	currState := c.getCurrentState()
-	if len(cluster.Spec.NodeManagerFeatures) > 0 {
+	if len(cluster.Spec.NodeManagerFeatures) > 0 && !currState {
 		err := b.Apply(ctx)
 
 		c.setCurrentState(err == nil)
-
 		return nil, err
+	} else if len(cluster.Spec.NodeManagerFeatures) > 0 && currState {
+		if err := c.setNodeManagerReplicas(ctx, obj, log, 0); err != nil {
+			return nil, err
+		}
+		if err := b.Apply(ctx); err != nil {
```

Comment:
Do we need to ensure the two scaling doesn't do concurrency issue? I mean do we need to wait a bit here?

Reply:
I had a `waitFor` function to do just that, but it seemed to test fine without it too so I removed as it seemed like overkill... then again with multiple nodes it might be necessary 🤔 I can add it again, it won't do any harm anyway I guess WDYT?

Reply:
If it works go ahead with this, we have time to fix it later if needed.

Reply:
Oh, just saw this now.. already added the change 🤣